rclpy/rclpy/executors.py

```diff
@@ -284,16 +284,16 @@ def wait_for_ready_callbacks(self, timeout_sec=None, nodes=None):
             clients = []
             services = []
             for node in nodes:
-                subscriptions.extend(self._filter_eligible_entities(node.subscriptions))
-                timers.extend(self._filter_eligible_entities(node.timers))
-                clients.extend(self._filter_eligible_entities(node.clients))
-                services.extend(self._filter_eligible_entities(node.services))
-                node_guards = self._filter_eligible_entities(node.guards)
+                subscriptions.extend(filter(self.can_execute, node.subscriptions))
+                timers.extend(filter(self.can_execute, node.timers))
+                clients.extend(filter(self.can_execute, node.clients))
+                services.extend(filter(self.can_execute, node.services))
+                node_guards = filter(self.can_execute, node.guards)
                 # retrigger a guard condition that was triggered but not handled
                 for gc in node_guards:
                     if gc._executor_triggered:
                         gc.trigger()
-                guards.extend(node_guards)
+                    guards.append(gc)
```

Comment:
I would be surprised if this is faster, and it appears to be exactly the same behavior. Do you have some reference which supports why this might be considered faster?

Reply:
It's a consequence of line 291 where `node_guards` becomes a generator instead of a list.

An alternative that uses `extend` instead of `append` is:

```python
                node_guards = list(filter(self.can_execute, node.guards))
                # retrigger a guard condition that was triggered but not handled
                for gc in node_guards:
                    if gc._executor_triggered:
                        gc.trigger()
                guards.extend(node_guards)
```

But that increases the executor overhead from 22.2% to 22.6% on my machine. I'm guessing the overhead of calling `append` for each item in the list is less than the overhead of creating an intermediate list and iterating internal to `extend`.

Reply:
I see so the removed line was implicitly `guards.extend(list(node_guards))`. 👍